File: web-auth-service/src/api/manage.js
import request from '../utils/request'

export const manageApi = {
  // 获取设备列表
  getDevices: () => request.get('/manage/devices/list'),
  
  // 踢出设备
  kickDevice: (deviceId) => request.post('/manage/devices/kick', { device_id: deviceId }),
  
  // 退出manage应用
  logout: () => request.post('/manage/devices/logout'),
  
  // SSO全局退出
  ssoLogout: () => request.post('/manage/devices/sso-logout'),
  
  // 退出所有设备
  logoutAll: () => request.post('/manage/devices/logout-all'),
  
  // 获取操作日志
  getLogs: (params) => request.get('/manage/logs', { params }),
  
  // 获取用户信息
  getProfile: () => request.get('/manage/profile')
}

File: server-auth-service/internal/router/manage.go
package router

import (
	"auth-service/internal/api"
	"auth-service/internal/middleware"

	"github.com/gin-gonic/gin"
)

type ManageRouter struct{}

func (m *ManageRouter) InitManageRouter(apiGroup *gin.RouterGroup) {
	// 需要鉴权的接口
	authenticated := apiGroup.Group("", middleware.AuthMiddleware())
	{
		manage := authenticated.Group("/manage")
		{
			manageApi := api.ApiGroupApp.ManageApi

			// 设备管理
			devices := manage.Group("/devices")
			{
				devices.GET("list", manageApi.GetDevices)
				devices.POST("/kick", manageApi.KickDevice)
				devices.POST("/logout", manageApi.Logout)
				devices.POST("/sso-logout", manageApi.SSOLogout)
				devices.POST("/logout-all", manageApi.LogoutAllDevices)
			}

			// 日志和用户信息
			manage.GET("/logs", manageApi.GetLogs)
			manage.GET("/profile", manageApi.GetProfile)
		}
	}
}
